internal/engine/execution/queries.go

```diff
@@ -150,3 +166,28 @@ func deleteSchema(ctx context.Context, tx sql.TxMaker, dbid string) error {
 
 	return sp.Commit(ctx)
 }
+
+// setContextualVars sets the contextual variables for the given postgres session.
+func setContextualVars(ctx context.Context, db sql.DB, data *common.ExecutionData) error {
+	_, err := db.Execute(ctx, fmt.Sprintf(`SET %s.%s = '%s';`, pgSessionPrefix, callerVar, data.Caller))
```

Comment:
I was wondering why we want a session variable instead of `SET LOCAL` so that there's no contamination or artifacts once the db transaction that wraps the procedure call commits/rollsback.

Also, can a procedure be called read-only / from RPC rather than in a consensus transaction?

Reply:
> I was wondering why we want a session variable instead of SET LOCAL so that there's no contamination or artifacts once the db transaction that wraps the procedure call commits/rollsback.

I did not know SET LOCAL was a thing. Good call.

> Also, can a procedure be called read-only / from RPC rather than in a consensus transaction?

Yes. It should set the caller if one is included (which is only enforced if kgw is used), and the txid will be set to a blank string.